ios/tunnel/tunnel_lockdown.go

```diff
@@ -87,21 +89,38 @@ func forwardTUNToDevice(ctx context.Context, mtu uint64, tun io.Reader, deviceCo
 	}
 }
 
-func forwardTCPToInterface(ctx context.Context, deviceConn io.Reader, tun io.Writer) error {
-	b := make([]byte, 20000)
-	for {
+func forwardTCPToInterface(ctx context.Context, mtu uint64, deviceConn io.Reader, tun io.Writer) error {
+	payload := make([]byte, mtu)
+	ip6Header := make([]byte, 40)
+
+	br := bufio.NewReader(deviceConn)
+	bw := bufio.NewWriter(tun)
 
+	for {
 		select {
 		case <-ctx.Done():
 			return nil
 		default:
-			n, err := deviceConn.Read(b)
+			_, err := io.ReadFull(br, ip6Header)
+			if err != nil {
+				return fmt.Errorf("failed to read IPv6 header: %w", err)
+			}
+
+			if ip6Header[0] != 0x60 {
```

Comment:
@dmissmann  I think IPv4 can happen here, no big deal and should not need an error

Reply:
Currently that shouldn't be possible, we configure the TUN interface only with an IPv6 https://github.com/danielpaulus/go-ios/blob/main/ios/tunnel/tunnel.go#L184

Reply:
hmm ok, but I definitely got some ipv4 packets on mac os when testing this.

Reply:
I might add code to just drop ipv4 then later.